### PR TITLE
CI: cancel old PR builds when new commits are pushed to a PR branch

### PR DESCRIPTION
**Pull request by DilumAluthge**
This leads to better resource utilization.

### CODE REVIEW

**Comment by DilumAluthge** [comment]
@omus Two CI jobs failed on Julia nightly with segfaults. Is this a known issue?

**Comment by DilumAluthge** [comment]
Alright, after a retry, all jobs are passing.

@omus This is ready to review.

**Comment by DilumAluthge** [comment]
Bump @omus

**Comment by omus** [comment]
Sorry for the slow response. Looking at this now.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -8,6 +8,11 @@ on:
     tags: "*"
   schedule:
     - cron: "0 2 * * *"  # Daily at 2 AM UTC (8 PM CST)
+concurrency:
+  # Skip intermediate builds: all builds except for builds on the `master` branch
+  # Cancel intermediate builds: only pull request builds
```
Comment (omus) [inline .github/workflows/CI.yml:13]:
```suggestion
# Skip/cancel execution of queued/running jobs from outdated commits. Jobs run on the
# `master` branch are not subject to these restrictions and are always executed.
concurrency:
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -8,6 +8,11 @@ on:
...
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.ref != 'refs/heads/master' || github.run_number }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
```
Comment (omus) [inline .github/workflows/CI.yml:15]: I can't think of a good reason not to cancel in progress jobs for branches and tags. Is there a scenario I'm not thinking of?
```suggestion
  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
```

**Review by omus — CHANGES_REQUESTED** [review]
Overall I'm good with the concurrency suggestion. Just have a couple of minor suggestions.

**Review by omus — APPROVED** [review]

**Pull request merged**